pkg/stanza/operator/input/namedpipe/watcher.go

```diff
@@ -0,0 +1,61 @@
+package namedpipe
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// Watcher watches a file for writes, notifying via `C` when one is observed.
+type Watcher struct {
+	C       chan struct{}
+	watcher *fsnotify.Watcher
+}
+
+// NewWatcher creates a new watcher for the given path.
+func NewWatcher(path string) (*Watcher, error) {
+	watcher, err := newWatcher(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Watcher{
+		C:       make(chan struct{}),
+		watcher: watcher,
+	}, nil
+}
+
+// Watch starts the watcher, sending a message to `C` when a write is observed.
+func (w *Watcher) Watch(ctx context.Context) error {
+	defer func() {
+		w.watcher.Close()
+		close(w.C)
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case event := <-w.watcher.Events:
+			if event.Op&fsnotify.Write == fsnotify.Write {
+				w.C <- struct{}{}
+			}
+		case err := <-w.watcher.Errors:
+			return fmt.Errorf("watcher error: %w", err)
+		}
+	}
+}
+
+func newWatcher(path string) (*fsnotify.Watcher, error) {
+	watcher, err := fsnotify.NewWatcher()
```

Comment:
Do you know if named pipes may be subject to this problems describe [here](github.com/fsnotify/fsnotify)?

Reply:
Not sure that link is working?

Reply:
Sorry, not sure what happened there. I've fixed the link.

Reply:
Cool. Yeah - that doesn't _really_ apply to us. The worry there is that files get rotated out from under the watcher, however here we assume that the collector will manage the pipe, and leave it undefined as to the behavior if something else fiddles with it